src/test/zcl_abapgit_gui_page_template.clas.abap

```diff
@@ -0,0 +1,131 @@
+CLASS zcl_abapgit_gui_page_template DEFINITION
+  PUBLIC
+  FINAL
+  INHERITING FROM zcl_abapgit_gui_component
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    INTERFACES:
+      zif_abapgit_gui_page_title,
+      zif_abapgit_gui_event_handler,
+      zif_abapgit_gui_hotkeys,
+      zif_abapgit_gui_menu_provider,
+      zif_abapgit_gui_renderable,
+      zif_abapgit_html_table.
+
+    CLASS-METHODS create
+*      IMPORTING
+*        ii_repo TYPE REF TO zif_abapgit_repo
+      RETURNING
+        VALUE(ri_page) TYPE REF TO zif_abapgit_gui_renderable
+      RAISING
+        zcx_abapgit_exception.
+
+    METHODS constructor
+*      IMPORTING
+*        ii_repo TYPE REF TO zif_abapgit_repo
+      RAISING
+        zcx_abapgit_exception.
+
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+
+    CONSTANTS:
+      BEGIN OF c_action,
+        " List of in-page actions
+        refresh TYPE string VALUE 'refresh',
+      END OF c_action.
+
+    CONSTANTS c_title TYPE string VALUE 'Page Template'. " Page title
+
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_GUI_PAGE_TEMPLATE IMPLEMENTATION.
+
+
+  METHOD constructor.
+    super->constructor( ).
+    " ...
+  ENDMETHOD.
+
+
+  METHOD create.
+
+    DATA lo_component TYPE REF TO zcl_abapgit_gui_page_template.
+
+    CREATE OBJECT lo_component.
+
+    ri_page = zcl_abapgit_gui_page_hoc=>create( lo_component ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_gui_event_handler~on_event.
+
+    CASE ii_event->mv_action.
+      WHEN c_action-refresh.
+        " ...
+        rs_handled-state = zcl_abapgit_gui=>c_event_state-re_render.
+      WHEN 'xyz'.
+        ASSERT 1 = 1.
+    ENDCASE.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_gui_hotkeys~get_hotkey_actions.
+
+    DATA ls_hotkey_action LIKE LINE OF rt_hotkey_actions.
+
+    ls_hotkey_action-ui_component = c_title.
+
+    ls_hotkey_action-description = 'Refresh'.
```

Comment:
It is example code, I'd suggest keeping it. 

hotkey as a part of menubar - a good idea but not that straight forward, that would be less flexible. hotkeys can be a part of the toolbar, a part of ANOTHER toolbar, and freely defined for whatever reason. Coupling them together is not ideal. What would be cool is automating creation of hotkeys from a toolbar. So that such a method would be smth like:

```
ri_hotkeys = get_main_toolbar->hotkeys( ).
ri_hotkeys->append( get_2nd_toolbar->hotkeys( ) ).
ri_hotkeys->add( key = 'x' title = 'Stuff' cmd = 'do-smth' ).
```

Reply:
yes, just like that 👍